torch/csrc/autograd/python_engine.cpp

```diff
@@ -23,14 +23,28 @@ struct THPEngine {
     PyObject_HEAD
 };
 
-static torch::autograd::python::PythonEngine engine;
+static bool _reinitialize_engine = false;
+
+namespace torch { namespace autograd { namespace python {
+
+PythonEngine::PythonEngine() = default;
 
-static Engine& get_python_engine() {
+Engine& PythonEngine::get_python_engine() {
+  static PythonEngine engine;
```

Comment:
Does this have the same behavior as the current code wrt when the Engine is actually created?
Or this will wait until it is first accessed before creating it?

Reply:
`PythonEngine` instance will not be created until first call to `get_python_engine()` (if compile with C++11 or later)

Reply:
Ok, given that we don't do anything in the constructor that should be fine as a change.